tarodb/assets_store.go

```diff
@@ -1226,6 +1254,30 @@ func queryChainAssets(ctx context.Context, q ActiveAssetsStore,
 	return matchingAssets, nil
 }
 
+// FetchCommitment returns a specific commitment identified by the given asset
+// parameters. If no commitment is found, ErrNoCommitment is returned.
+func (a *AssetStore) FetchCommitment(ctx context.Context, id asset.ID,
```

Comment:
Any reason this was added instead of using `SelectCommitment` which is more generic? Alternatively if a simpler API is needed elsewhere, then this can just call directly into that.

Reply:
The `SelectCommitment` didn't allow us to specify the asset we want precisely enough. There was always the chance that we'd get the wrong one if we had multiple asset UTXOs of the same value. I fixed that by adding the `FetchCommitment` method which now also takes the script key to make an asset UTXO identifiable.